azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/DeploymentToNodeMetadata.java

```diff
@@ -0,0 +1,162 @@
+      if (text != null) {
+         for (Deployment.ProvisioningState status : Deployment.ProvisioningState.values()) {
+            if (text.equalsIgnoreCase(status.name())) {
+               return status;
+            }
+         }
+      }
+      return Deployment.ProvisioningState.UNRECOGNIZED;
+   }
+
+   private final AzureComputeApi api;
+
+   private final Supplier<Set<? extends Location>> locations;
+
+   private final GroupNamingConvention nodeNamingConvention;
+
+   private final VMImageToImage vmImageToImage;
+
+   private final VMHardwareToHardware vmHardwareToHardware;
+
+   private final Map<String, Credentials> credentialStore;
+
+   @Inject
+   DeploymentToNodeMetadata(
+           AzureComputeApi api,
+           @Memoized Supplier<Set<? extends Location>> locations,
+           GroupNamingConvention.Factory namingConvention, VMImageToImage vmImageToImage,
+           VMHardwareToHardware vmHardwareToHardware, Map<String, Credentials> credentialStore) {
+
+      this.nodeNamingConvention = namingConvention.createWithoutPrefix();
+      this.locations = locations;
+      this.vmImageToImage = vmImageToImage;
+      this.vmHardwareToHardware = vmHardwareToHardware;
+      this.credentialStore = credentialStore;
+      this.api = api;
+   }
+
+   @Override
+   public NodeMetadata apply(final VMDeployment from) {
+      final NodeMetadataBuilder builder = new NodeMetadataBuilder();
+      final Deployment deployment = from.deployment;
+      builder.id(deployment.name());
+      builder.providerId(deployment.name());
+      builder.name(deployment.name());
+      String group = deployment.name();
+      int index = group.lastIndexOf("-");
+      builder.group(group.substring(0, index));
+
+      NodeMetadata.Status status = STATUS_TO_NODESTATUS.get(provisioningStateFromString(deployment.properties().provisioningState()));
+      if (status == NodeMetadata.Status.RUNNING && from.vm != null && from.vm.statuses() != null) {
+         List<VirtualMachineInstance.VirtualMachineStatus> statuses = from.vm.statuses();
+         for (int c = 0; c < statuses.size(); c++) {
+            if (statuses.get(c).code().substring(0, 10).equals("PowerState")) {
+               if (statuses.get(c).displayStatus().equals("VM running")) {
+                  status = NodeMetadata.Status.RUNNING;
+               } else if (statuses.get(c).displayStatus().equals("VM stopped")) {
+                  status = NodeMetadata.Status.SUSPENDED;
+               }
+               break;
+            }
+         }
+      }
+
+      builder.status(status);
+
+      Credentials credentials = credentialStore.get("node#" + from.deployment.name());
+      if (credentials != null && credentials.identity.equals(JCLOUDS_DEFAULT_USERNAME)) {
+         credentials = new Credentials(AZURE_DEFAULT_USERNAME, credentials.credential);
+      } else if (credentials == null) {
+         credentials = new Credentials(AZURE_DEFAULT_USERNAME, DEFAULT_LOGIN_PASSWORD);
+      }
```

Comment:
I can't remember all the details now but we needed to do this. At least jClouds default username&password are not suitable with Azure. For example Azure does not allow to use "root" username.